src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -300,10 +323,12 @@ private void TrySetRecoverableError(SyntaxDiagnosticInfo error)
             }
 
             internal void ScanInterpolatedStringLiteralTop(
+                out InterpolatedStringKind kind,
                 out Range openQuoteRange,
                 ArrayBuilder<Interpolation>? interpolations,
                 out Range closeQuoteRange)
             {
+                kind = _kind;
```

Comment:
It seems fine if this is the most convenient way to pass along and use the `kind`, but it might also make sense to use a factoring that makes it more clear that this value is decided before the call even occurs, such as by putting this into an accessible property or calculating it and passing as a parameter to the `InterpolatedStringScanner`.

Reply:
gotcha.  let me see what i can do.

Reply:
see what you think now.

Reply:
i rolled back my change.  it made later stuff in teh raw-interpolated branch harder IMO.

Reply:
i rolled back my rollback.  now i like it again (and the raw-interpolated branch follows taht pattern).